e2e/recipes/recipes.go

```diff
@@ -0,0 +1,130 @@
+// SPDX-FileCopyrightText: The RamenDR authors
+// SPDX-License-Identifier: Apache-2.0
+
+package recipes
+
+import (
+	recipe "github.com/ramendr/recipe/api/v1alpha1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"github.com/ramendr/ramen/e2e/config"
+	"github.com/ramendr/ramen/e2e/types"
+)
+
+const (
+	// Constant serves as the default timeout for the resources to come up when executing hook.
+	defaultHookTimeout = 300
+	resource           = "resource"
+)
+
+func Generate(ctx types.TestContext, recipeConfig *config.Recipe) *recipe.Recipe {
+	ctxName := ctx.Name()
+	appNamespace := ctx.AppNamespace()
+
+	return &recipe.Recipe{
+		TypeMeta: metav1.TypeMeta{
+			Kind:       "Recipe",
+			APIVersion: "recipe.ramendr.io/v1alpha1",
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      ctxName,
+			Namespace: appNamespace,
+		},
+		Spec: recipe.RecipeSpec{
+			Hooks:     generateHooks(ctx, recipeConfig),
+			Groups:    generateGroups(ctx, appNamespace),
+			Workflows: generateWorkflows(recipeConfig),
+		},
+	}
+}
+
+func generateHooks(ctx types.TestContext, rc *config.Recipe) []*recipe.Hook {
+	var hooks []*recipe.Hook
+
+	if rc.CheckHook {
+		checkHook := prepareCheckHook(ctx)
+		hooks = append(hooks, checkHook)
+	}
+
+	if rc.ExecHook {
+		execHook := prepareExecHook(ctx)
+		hooks = append(hooks, execHook)
+	}
+
+	return hooks
+}
+
+// XXX document in e2e.doc need for RBAC for specific resources
+func prepareCheckHook(ctx types.TestContext) *recipe.Hook {
+	return &recipe.Hook{
+		Name:           "check-hook",
+		Type:           "check",
+		Namespace:      ctx.AppNamespace(),
+		LabelSelector:  ctx.Workload().GetLabelSelector(),
+		SelectResource: ctx.Workload().GetSelectResource(),
+		Timeout:        defaultHookTimeout,
+		Chks: []*recipe.Check{
+			{
+				Name:      "check-replicas",
+				Condition: "{$.spec.replicas} == {$.status.readyReplicas}", // XXX works only for resources with .replicas
```

Comment:
Only works for resources with replicas means we must fail if you specify SelectResource that does not have replicas. We need to resolve this now.